tests/geometries/helpers/geometries.ts

```diff
@@ -35,3 +55,80 @@ export const generatePointInput = ({ bBox, coordinates, crs }: GenerateGeometryI
     coordRefSys,
   });
 };
+
+export const generateLineInput = ({
+  bBox,
+  coordinates,
+  crs,
+}: GenerateGeometryInput<LineStringInput['coordinates']> = {}): fc.Arbitrary<LineStringInput> => {
+  const lineInput = (
+    coordinates?.chain((positions) => {
+      return fc.constant(
+        positions.map(([east, north]) => {
+          return [east, north];
+        })
+      );
+    }) ??
+    (bBox
+      ? bBox.chain(([minEast, minNorth, maxEast, maxNorth]) =>
+          fc.array(
+            fc.tuple(
+              fc.float({ min: Math.fround(minEast), max: Math.fround(maxEast), noNaN: true, minExcluded: true, maxExcluded: true }),
+              fc.float({ min: Math.fround(minNorth), max: Math.fround(maxNorth), noNaN: true, minExcluded: true, maxExcluded: true })
+            ),
+            { minLength: 2 }
+          )
+        )
+      : fc.array(fc.tuple(fc.float({ noDefaultInfinity: true, noNaN: true }), fc.float({ noDefaultInfinity: true, noNaN: true })), { minLength: 2 }))
+  ).chain((coordinates) =>
+    fc.record({
+      coordinates: fc.constant(coordinates),
+      bbox: fc.option(fc.constant(coordinatesToBBox(coordinates)), { nil: undefined }),
+      coordRefSys: fc.option(crs ?? fc.constantFrom(...SUPPORTED_CRS), { nil: undefined }),
+    })
+  );
+
+  return lineInput;
+};
+
+// Notice: currently does not generates holes
+export const generatePolygonInput = ({
```

Comment:
Please add a constraint so we could also generate actually valid polygons (first and last point are equal).

Reply:
done